src/util.rs

```diff
@@ -44,6 +44,28 @@ pub fn history() -> web_sys::History {
     window().history().expect("Can't find history")
 }
 
+/// Convenience function to access the `web_sys::HtmlDocument`.
+pub fn html_document() -> web_sys::HtmlDocument {
+    wasm_bindgen::JsValue::from(document()).unchecked_into::<web_sys::HtmlDocument>()
+}
+
+/// Convenience function to get all cookies from the current `HtmlDocument`
+pub fn cookies() -> Option<cookie::CookieJar> {
+    html_document().cookie().ok().and_then(|cookie_str| {
+        let cookies = cookie_str
+            .split(";")
+            .map(|x| x.to_owned())
+            .filter_map(|cs| cookie::Cookie::parse_encoded(cs).ok());
+        if cookies.clone().count() > 0 {
```

Comment:
Maybe a little bit faster / readable code:
```rust
/// Convenience function to get all cookies from the current `HtmlDocument`
/// _Note:_ Returns `None` if parsing cookies fails or there are no cookies.
pub fn cookies() -> Option<cookie::CookieJar> {
    let cookies_str = html_document().cookie().ok()?;
    let mut jar = cookie::CookieJar::new();

    for cookie_str in cookies_str.split(';') {
        let cookie = cookie::Cookie::parse_encoded(cookie_str).ok()?;
        jar.add(cookie.into_owned());
    }

    let jar_is_empty = jar.iter().next().is_none();
    if jar_is_empty {
        None
    } else {
        Some(jar)
    }
}
```
Notes:
- It also returns `None` if it fails to parse cookies.
- I assume that creating a new jar instance is a trivial operation.